TFNetworkLayer.py

```diff
@@ -3345,12 +3345,19 @@ def __init__(self, axis, dim=1, **kwargs):
     data = self.input_data
     if isinstance(axis, int):
       data = data.copy_as_batch_major()
-    axis = self._get_axis(data=data, axis=axis)
+    axis_index = self._get_axis(data=data, axis=axis)
     from TFUtil import expand_dims_unbroadcast
-    self.output.placeholder = expand_dims_unbroadcast(data.placeholder, axis=axis, dim=dim)
+    self.output.placeholder = expand_dims_unbroadcast(data.placeholder, axis=axis_index, dim=dim)
     self.output.size_placeholder = {
-      (i if (data.get_batch_axis(i) < axis) else i + 1): v
+      (i if (data.get_batch_axis(i) < axis_index) else i + 1): v
       for (i, v) in data.size_placeholder.items()}
+    if axis in ["spatial", "time", "T"]:
+      self.output.time_dim_axis = axis_index
+
+      # The sequence length of the time axis is know to be 1 at this point. But subsequent layers (eg. HDFDumpLayer)
+      # may expect the sequence length to be variable
```

Comment:
No, this should not be needed. This should be fixed in `HDFDumpLayer`, or other layers which cannot handle that correctly.